mlx/backend/metal/utils.h

```diff
@@ -123,6 +124,31 @@ MTL::Size get_block_dims(int dim0, int dim1, int dim2) {
   return MTL::Size{1ul << pows[0], 1ul << pows[1], 1ul << pows[2]};
 }
 
+NS::String* make_string(std::ostringstream& os) {
+  return NS::String::string(os.str().c_str(), NS::UTF8StringEncoding);
```

Comment:
I thought `str()` was a temporary so it might be safer to get the `str` in the line above so you can be sure the constructor completes safely?

Reply:
`NS::String::string` calls [`initWithCString:encoding:`](https://developer.apple.com/documentation/foundation/nsstring/1411950-initwithcstring?language=objc), which copies the C string.

Reply:
Ah, makes sense now. Thanks for elaborating.